### PR TITLE
tests: updated names for the tests

### PR DESCRIPTION
**Pull request by maksimmasalski**
Some test cases have the same test case name.
To get rid of it, I decided to change test cases names
contained same names.
Please check my logic, how I give them names.
Usually trying to give name same as a directory folder.
There are still more test cases which necessary to change,
but I will make changes by small steps.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

### CODE REVIEW

File: tests/lib/json/testcase.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 tests:
-  libraries.encoding:
+  libraries.json.encoding:
```
Comment (nashif) [inline tests/lib/json/testcase.yaml]: libraries.encoding.json

File: tests/subsys/jwt/testcase.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 tests:
-  libraries.encoding:
+  libraries.subsys.jwt.encoding:
```
Comment (nashif) [inline tests/subsys/jwt/testcase.yaml]: libraries.encoding.jwt

File: tests/unit/base64/testcase.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 tests:
-  libraries.encoding:
+  libraries.unit.base.encoding:
```
Comment (nashif) [inline tests/unit/base64/testcase.yaml]: libraries.encoding.base64

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by pfalcon — APPROVED** [review]
Changes to POSIX tests look ok.

**Comment by maksimmasalski** [comment]
> Changes to POSIX tests look ok.

Hi. I can't understand why test is crashing. I reverted changes, and still the same error.

**Comment by pfalcon** [comment]
>>    Changes to POSIX tests look ok.

> Hi. I can't understand why test is crashing. I reverted changes, and still the same error.

D'oh, so it's a POSIX test which is failing... Well, as you can see in ffb9e0977da03bdb4563ca73b01a65588687c2c8, a check just above the currently failing was already adjusted to use `>=` instead of just `>`. Dunno why 2nd case wasn't adjusted at that time, feel free to do it now (as a separate commit of course).

**Comment by maksimmasalski** [comment]
> > > Changes to POSIX tests look ok.
> 
> > Hi. I can't understand why test is crashing. I reverted changes, and still the same error.
> 
> D'oh, so it's a POSIX test which is failing... Well, as you can see in [ffb9e09](https://github.com/zephyrproject-rtos/zephyr/commit/ffb9e0977da03bdb4563ca73b01a65588687c2c8), a check just above the currently failing was already adjusted to use `>=` instead of just `>`. Dunno why 2nd case wasn't adjusted at that time, feel free to do it now (as a separate commit of course).

#20048 I hope I understand it right.

**Review by nashif — APPROVED** [review]

**Pull request merged**